test/smoke/dashCore.js

```diff
@@ -1,18 +1,107 @@
-describe('DashCore', () => {
-  describe('All nodes', () => {
-    // `subversion` from `getNetworkInfo` response.
-    it('should have correct network type');
-
-    // The number of blocks should be almost the same (-3/+3) and block hash of particular
-    // block height should be the same. Using `blocks` and `bestblockhash` from `GetBlockChainInfo`.
-    it('should propagate blocks');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+describe('All nodes', () => {
+  let nodeNames = networkConfig.inventory.masternodes.hosts;
+  nodeNames = nodeNames.concat(networkConfig.inventory['wallet-nodes'].hosts);
+  // TODO miner rpc call stuck. why?
+  // nodeNames = nodeNames.concat(networkConfig.inventory['miners'].hosts);
```

Comment:
still stuck?

Reply:
await rpc.waitForNewBlock(1) doesn't work now.
for example the following code

     const { result } = await rpc.getBlockCount();
      const { result: { height } } = await rpc.waitForNewBlock(1);
      expect(result + 1).to.be.equal(height);
doesn't wait new block. and rpc.waitForNewBlock returns height=result immediately